src/componentprocessor/index.ts

```diff
@@ -0,0 +1,142 @@
+import * as fs from "fs";
+import * as path from "path";
+import { promisify } from "util";
+import xmldoc, { XmlDocument } from "xmldoc";
+import pSettle = require("p-settle");
+const readFile = promisify(fs.readFile);
+import * as fg from "fast-glob";
+
+interface FieldAttributes {
+    id: string;
+    type: string;
+    alias?: string;
+    value?: string;
+    onChange?: string;
+    alwaysNotify?: string;
+}
+
+interface ComponentField {
+    [key: string]: FieldAttributes;
+}
+
+export class ComponentDefinition {
+    public contents?: string;
+    public xmlNode?: XmlDocument;
+    public name?: string;
+    // indicates whether this component hierarchy has been processed before
+    // which means the fields, children, and inherited functions are correctly set
+    public processed: boolean = false;
+    public fields: ComponentField = {};
+
+    constructor(readonly xmlPath: string) {}
+
+    async parse(): Promise<ComponentDefinition> {
+        let contents;
+        try {
+            contents = await readFile(this.xmlPath, "utf-8");
+            let xmlStr = contents.toString().replace(/\r?\n|\r/g, "");
+            this.xmlNode = new xmldoc.XmlDocument(xmlStr);
+            this.name = this.xmlNode.attr.name;
+
+            return Promise.resolve(this);
+        } catch (err) {
+            // TODO: provide better parse error reporting
+            //   cases:
+            //     * file read error
+            //     * XML parse error
+            return Promise.reject(this);
```

Comment:
Would printing out the `err` at least provide some info on what went wrong?

Reply:
It gets thinks a bit messy when running the unit tests because one of those checks for invalid xml syntax

Reply:
Let's revisit this in the future possibly having an early failure case to check all XML files.